crates/autopilot/src/limit_order_quoter.rs

```diff
@@ -0,0 +1,101 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use chrono::Duration;
+use ethcontract::U256;
+use model::{
+    order::{Order, OrderKind, OrderUid},
+    quote::{default_verification_gas_limit, OrderQuoteSide, QuoteSigningScheme, SellAmount},
+    signature::Signature,
+};
+use shared::order_quoting::{OrderQuoting, QuoteParameters};
+use std::sync::Arc;
+
+#[async_trait]
+pub trait LimitOrderFeeStoring: Send + Sync {
+    async fn limit_orders_with_outdated_fees(&self, age: Duration) -> Result<Vec<Order>>;
+    async fn update_surplus_fee(&self, order_uid: &OrderUid, surplus_fee: U256) -> Result<()>;
+}
+
+/// Background task which quotes all limit orders and sets the surplus_fee for each one
+/// to the fee returned by the quoting process. If quoting fails, the corresponding
+/// order is filtered out.
+pub struct LimitOrderQuoter {
+    pub limit_order_age: Duration,
+    pub loop_delay: std::time::Duration,
+    pub quoter: Arc<dyn OrderQuoting>,
+    pub storage: Arc<dyn LimitOrderFeeStoring>,
+}
+
+impl LimitOrderQuoter {
+    pub fn spawn(self) {
+        tokio::spawn(async move {
+            loop {
+                if let Err(err) = self.update().await {
+                    tracing::warn!(?err, "failed to update limit order surplus");
```

Comment:
I guess for alerting we either need `tracing::error!()` or a metric to configure alerts through Grafana.

Reply:
https://github.com/cowprotocol/services/pull/714/commits/915cf8cae2d5ef40869b08c5643fe40f67b72528